tracer/src/Datadog.Trace/Logging/DirectSubmission/DirectLogSubmissionManager.cs

```diff
@@ -0,0 +1,63 @@
+﻿// <copyright file="DirectLogSubmissionManager.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Logging.DirectSubmission.Formatting;
+using Datadog.Trace.Logging.DirectSubmission.Sink;
+using Datadog.Trace.Logging.DirectSubmission.Sink.PeriodicBatching;
+
+namespace Datadog.Trace.Logging.DirectSubmission
+{
+    internal class DirectLogSubmissionManager : IDisposable
+    {
+        private static readonly IDatadogLogger Logger = DatadogLogging.GetLoggerFor<DirectLogSubmissionManager>();
+
+        private DirectLogSubmissionManager(DirectLogSubmissionSettings settings, IDatadogSink sink, LogFormatter formatter)
+        {
+            Settings = settings;
+            Sink = sink;
+            Formatter = formatter;
+        }
+
+        public DirectLogSubmissionSettings Settings { get; }
+
+        public IDatadogSink Sink { get; }
+
+        public LogFormatter Formatter { get; }
+
+        public static DirectLogSubmissionManager Create(
+            DirectLogSubmissionSettings settings,
+            string serviceName,
+            string env,
+            string serviceVersion)
+        {
+            var formatter = new LogFormatter(settings, serviceName, env, serviceVersion);
+            if (!settings.IsEnabled)
+            {
+                return new DirectLogSubmissionManager(settings, new NullDatadogSink(), formatter);
+            }
+
+            // TODO: create batching options from config? Or should we not expose these options, and stick to the defaults?
+            var batchingOptions = new BatchingSinkOptions();
```

Comment:
I do think it would be good to allow these to be setup from config if users need to lower pressure or are willing to handle more.

Reply:
Added ability to configure these in https://github.com/DataDog/dd-trace-dotnet/pull/1945/commits/c26881c0c79208837097cef41f388c94baeffb99